[iota-analytics-indexer] Add README.md

tomxey opened the pull request:
# Description of change

Add README.md for iota-analytics-indexer

## Links to any relevant issues

#3976

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation Fix

## How the change has been tested

N.A.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

tomxey committed:
Add README.md

kodemartin commented:
Had a first quick read:

* I don't get what metrics are collected though? How does it compare to the `indexer-analytical-worker`?
* Could you elaborate on the `BigQuery/Snowflake` schemas used?
* What is `BigQuery` and `Snowflake` in the first place? It would be helpful to write a few things and provide references links for digging deeper.

tomxey committed:
Fixing review remarks:
 - Added more info about SnowFlake and BigQuery
 - Expanded a bit the comparison with iota-indexer
 - Added more info about schemas for exported data

tomxey commented:
> Had a first quick read:
> 
>     * I don't get what metrics are collected though? How does it compare to the `indexer-analytical-worker`?
> 
>     * Could you elaborate on the `BigQuery/Snowflake` schemas used?
> 
>     * What is `BigQuery` and `Snowflake` in the first place? It would be helpful to write a few things and provide references links for digging deeper.

@kodemartin Pushed changes that are addressing those remarks.

kodemartin commented on crates/iota-analytics-indexer/README.md:
```diff
@@ -0,0 +1,125 @@
+
+ Those metrics are computed by a separate analytical worker instance of the indexer, but it uses the main DB as the main indexer instance.
+
+It seems that some of the values stored in main indexer tables by iota-indexer's `fullnode_sync_worker` are only stored there for analytical purposes (move calls, tx recipients) and could potentially be not processed/stored if not for analytical reasons. 
```
Nit: Phrasing
```suggestion
It seems that some of the values stored in main indexer tables by iota-indexer's `fullnode_sync_worker` are only stored there for analytical purposes (move calls, tx recipients) and could potentially be excluded from further processing if it were not for analytical reasons. 
```

tomxey replied:
done

kodemartin commented on crates/iota-analytics-indexer/README.md:
```diff
@@ -0,0 +1,125 @@
+
+The `iota-analytics-indexer` is not computing any analytical metrics directly.
+It is only exporting data for further processing via external tools (BigQuery/SnowFlake).
+Functionality from `iota-indexer` that is not required to serve user JSON RPC/GraphQL requests could potentially be moved away from `iota-indexer` and served by some other tool based on data exported by the `iota-analytics-indexer`.
```
Nit: Phrasing (if I get it right)
```suggestion
On this premise, the functionality  in `iota-indexer` that is currently used for extracting analytics (and thus unrelated to the JSON-RPC/GraphQL service)  could be moved out and delegated to another tool that processes data exported by `iota-analytics-indexer`.
```

tomxey replied:
Done

kodemartin commented on crates/iota-analytics-indexer/README.md:
```diff
@@ -0,0 +1,125 @@
+The `iota-analytics-indexer` is not computing any analytical metrics directly.
+It is only exporting data for further processing via external tools (BigQuery/SnowFlake).
+Functionality from `iota-indexer` that is not required to serve user JSON RPC/GraphQL requests could potentially be moved away from `iota-indexer` and served by some other tool based on data exported by the `iota-analytics-indexer`.
+The sync logic in `iota-indexer` could be simplified a bit then to store only data that is needed to serve user requests.
```
Nit:
```suggestion
Then the sync logic in `iota-indexer` could be simplified as well to store only data that is needed for the purposes of the RPC APIs.
```

tomxey replied:
Done

kodemartin commented on crates/iota-analytics-indexer/README.md:
```diff
@@ -0,0 +1,125 @@
+ - TRANSACTION
+
+
+> Note: The following rust structs currently do not have DB schemas prepared:
```
Nit: For further emphasis (see also [GH alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts))
```suggestion
> [!NOTE]
> The following rust structs currently do not have DB schemas prepared:
```

tomxey replied:
Thanks for the tip. Done.

kodemartin reviewed (approved):
Looks good @tomxey ! Left a few optional suggestions.

Also note that the diagram does not look very clear on dark mode (there are some hatch lines overlapping with text for example).

sergiupopescu199 reviewed (approved):
LGTM ✨

tomxey committed:
Fix review remarks about README contents

tomxey committed:
Applied automatic dprint format

tomxey committed:
Use standard mermaid diagram theme

tomxey committed:
Style diagram to look better in both dark and light modes

tomxey commented:
> Also note that the diagram does not look very clear on dark mode (there are some hatch lines overlapping with text for example).

@kodemartin Changed styling of the diagram. Now it should be readable both in dark and light modes.
